component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -294,7 +308,7 @@ private void addOrValidateCertificate(String userName, AuthenticationContext aut
             }
 
             // Check whether user account is disabled or not.
-            if (isAccountDisabled(getUsername(authenticationContext))) {
+            if (isAccountDisabled(AuthenticatedUser.createLocalAuthenticatedUserFromSubjectIdentifier(userName))) {
```

Comment:
Can't we use the available userName variable?

Reply:
Currently method is accepting an AuthenticatedUser object. In that case method signature and implementation have to be changed?

Reply:
ok then fine